java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexHeightData.java

```diff
@@ -101,22 +104,38 @@ private TileData(int id) {
 		public File loadData(String srtmDataUrl, File workDir) throws IOException {
 			dataLoaded = true;
 			File f = loadFile(getFileName() + ".tif", srtmDataUrl, workDir);
-			BufferedImage img;
 			if (f.exists()) {
-				try {
-					img = ImageIO.read(f);
+				boolean readSuccess = false;
+				try (ImageInputStream iis = ImageIO.createImageInputStream(f)) {
+					Iterator<ImageReader> readers = ImageIO.getImageReadersByFormatName("TIFF");
+					while (readers.hasNext() && !readSuccess) {
```

Comment:
Still the order for tif readers is not fixed. So it's actually spends CPU time to read with in approriate reader